sdks/java/core/src/test/java/org/apache/beam/sdk/io/CompressedSourceTest.java

```diff
@@ -235,6 +315,30 @@ public void testReadConcatenatedGzip() throws IOException {
     assertEquals(Bytes.asList(expected), actual);
   }
 
+  /**
+   * Using Lzo Codec Test a concatenation of lzo files is correctly decompressed.
+   *
+   * <p>A concatenation of lzo files as one file is a valid lzo file and should decompress to be the
+   * concatenation of those individual files.
+   */
+  @Test
+  public void testReadConcatenatedLzo() throws IOException {
```

Comment:
Please add a `testReadConcatenatedLzop()` as well if it is applicable. Not sure if it is due to headers.

Reply:
The current behaviour of LZOP codec is that it returns the contents of the first file only, if concatenated files are given because of the presence of headers. This causes the test to fail. That is why we have not added this test.

Reply:
Perhaps it would be a good idea to add a test with an expected failure then?

Reply:
We have added this in this update.

Reply:
Can we either add support for multistream or throw an exception if the stream isn't finished?

It would be dangerous for users to have part of their data silently dropped in this scenario. We should also add to the comment that concatenated streams aren't supported.